File: frontend/src/types/course.ts
export type SubLessonProgress = 'not-started' | 'in-progress' | 'completed'

export type AssignmentStatus = 'pending' | 'submitted'

export type Assignment = {
  id: string
  question: string
  status: AssignmentStatus
  answer?: string
  deadlineLabel: string
}

export type SubLesson = {
  id: string
  title: string
  description: string
  videoUrl: string
  progress: SubLessonProgress
  assignment?: Assignment
}

export type Module = {
  id: string
  title: string
  subLessons: SubLesson[]
}

export type Course = {
  id: string
  category: string
  title: string
  description: string
  longDescription: string
  imageUrl: string
  lessonCount: number
  hourCount: number
  price: number
  modules: Module[]
}

export type AdminCourseLesson = { id: number; name: string; subLessons: number }

export type AdminCourse = {
  id: number
  name: string
  lessons: number
  price: number
  createdAt: string
  updatedAt: string
  accent: string
  category?: string | null
  learningTime?: number | null
  hasPromo?: boolean
  promoCode?: string | null
  minimumPurchase?: number | null
  discount?: number | null
  discountType?: 'percentage' | 'fixed' | null
  summary?: string | null
  description?: string | null
  imageName?: string | null
  videoName?: string | null
  resourceName?: string | null
  lessonItems?: AdminCourseLesson[]
}

export type AdminCoursePayload = Omit<
  AdminCourse,
  'id' | 'lessons' | 'createdAt' | 'updatedAt' | 'lessonItems'
> & {
  lessonItems: Array<Omit<AdminCourseLesson, 'id'>>
}

File: backend/src/main/java/com/courseflow/course/Course.java
package com.courseflow.course;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

public record Course(
    long id,
    String name,
    int lessons,
    BigDecimal price,
    OffsetDateTime createdAt,
    OffsetDateTime updatedAt,
    String accent,
    String category,
    Integer learningTime,
    boolean hasPromo,
    String promoCode,
    BigDecimal minimumPurchase,
    BigDecimal discount,
    String discountType,
    String summary,
    String description,
    String imageName,
    String videoName,
    String resourceName,
    List<CourseLesson> lessonItems
) {
}
